controller/resource_roles.go

```diff
@@ -69,6 +82,62 @@ func (c *ResourceRolesController) ListAssignedByRoleName(ctx *app.ListAssignedBy
 	})
 }
 
+// AssignRole assigns a specific role for a resource, to one or more identities.
+func (c *ResourceRolesController) AssignRole(ctx *app.AssignRoleResourceRolesContext) error {
+
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+	if currentUser == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("identity ID not found in token"))
+	}
+
+	// check if the current user token belongs to a user who has the necessary privileges
+	// for assigning roles to other users.
+	hasScope, err := c.permissionService.HasScope(ctx, *currentUser, ctx.ResourceID, ROLE_ASSIGNMENT_SCOPE)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+	if !hasScope {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to assign roles"))
```

Comment:
I'm ok with using both 401 or 403 here. But we must be contestant with our error statuses when rejecting privileged operations because of missing permissions.

@sbryzak can you reminder what we are supposed to return here? 401 or 403?

Reply:
@alexeykazakov I did some quick refresher research on this, there is actually a problem with the 401 response in that the standard seems to incorrectly label is as 'Unauthorized' when in actual fact the error code is used to return an authentication error.  I.e. the user hasn't authenticated, or their authentication is incorrect somehow.  In this instead, I think @sbose78 's use of 403 is correct, as the user is definitely authenticated (otherwise they wouldn't have gotten this far) however they lack the scope necessary to perform the operation.  This is an easy mistake to make due to the confusion surrounding the http response codes, and I wouldn't be surprised if I've made it myself elsewhere in the code.

Reply:
I'm fine with 403. We just have to make sure we always use 403 if there is lack of permissions.